plugins/modules/files/iso_customize.py

```diff
@@ -0,0 +1,326 @@
+            MODULE_ISO_CUSTOMIZE.fail_json(msg=ret_msg)
+
+    try:
+        if ISO_MODE == "iso9660":
+            opened_iso.add_file(file_local, iso_path=file_in_iso_path)
+        elif ISO_MODE == "rr":
+            opened_iso.add_file(file_local, iso_path=file_in_iso_path, rr_name=file_name)
+        elif ISO_MODE == "joliet":
+            opened_iso.add_file(file_local, iso_path=file_in_iso_path, joliet_path=dest_file)
+        elif ISO_MODE == "udf":
+            opened_iso.add_file(file_local, iso_path=file_in_iso_path, udf_path=dest_file)
+    except Exception as err:
+        msg = "Failed to add local file %s to ISO with error: %s" % (file_local, to_native(err))
+        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+
+    return 0, ""
+
+
+def iso_delete_file(opened_iso, dest_file):
+    dest_file = dest_file.strip()
+    if dest_file[0] != "/":
+        dest_file = "/%s" % dest_file
+    file_name = os.path.basename(dest_file)
+
+    if '.' not in file_name:
+        file_in_iso_path = dest_file.upper() + '.;1'
+    else:
+        file_in_iso_path = dest_file.upper() + ';1'
+
+    try:
+        if ISO_MODE == "iso9660":
+            record = opened_iso.get_record(iso_path=file_in_iso_path)
+        elif ISO_MODE == "rr":
+            record = opened_iso.get_record(rr_path=dest_file)
+        elif ISO_MODE == "joliet":
+            record = opened_iso.get_record(joliet_path=dest_file)
+        elif ISO_MODE == "udf":
+            record = opened_iso.get_record(udf_path=dest_file)
+
+        if record and not record.is_file():
+            return -1, "The file %s does not exists in ISO or not a file." % dest_file
+    except Exception as err:
+        msg = "Failed to get record of file %s in ISO with filesystem %s. error: %s" % (
+            dest_file, ISO_MODE, to_native(err))
+        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+
+    try:
+        if ISO_MODE == "iso9660":
+            opened_iso.rm_hard_link(iso_path=file_in_iso_path)
+        elif ISO_MODE == "rr":
+            opened_iso.rm_hard_link(iso_path=file_in_iso_path)
+        elif ISO_MODE == "joliet":
+            opened_iso.rm_hard_link(joliet_path=dest_file)
+        elif ISO_MODE == "udf":
+            opened_iso.rm_hard_link(udf_path=dest_file)
+    except Exception as err:
+        msg = "Failed to delete iso file %s with error: %s" % (dest_file, to_native(err))
+        MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
+
+    return 0, ""
+
+
+def iso_rebuild(src_iso, dest_iso, delete_files_list, add_files_list):
+    global ISO_MODE
+
+    try:
+        iso = pycdlib.PyCdlib()
+        iso.open(src_iso)
+        if iso.has_rock_ridge():
+            ISO_MODE = "rr"
+        elif iso.has_joliet():
+            ISO_MODE = "joliet"
+        elif iso.has_udf():
+            ISO_MODE = "udf"
+
+        for item in delete_files_list:
+            ret, msg = iso_delete_file(iso, item)
+            if ret != 0:
+                MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)
```

Comment:
Why return a return value, instead of simply calling `MODULE_ISO_CUSTOMIZE.fail_json(msg=msg)` directly in the function?

Reply:
Fix it. Thanks